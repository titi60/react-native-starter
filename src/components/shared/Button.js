import React from 'react';
import { TouchableOpacity } from 'react-native';

const Button = ({ onPress, children }) => (
  <TouchableOpacity testID="button-touchable" onPress={onPress}>
    {children}
  </TouchableOpacity>
);

export default Button;
