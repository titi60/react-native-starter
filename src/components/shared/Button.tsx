import React from 'react';
import { TouchableOpacity } from 'react-native';

interface IProps {
  onPress: () => void;
  children?: Element;
  testID?: string;
}

const Button = ({ onPress, children }: IProps) => (
  <TouchableOpacity testID="button-touchable" onPress={onPress}>
    {children}
  </TouchableOpacity>
);

export default Button;
