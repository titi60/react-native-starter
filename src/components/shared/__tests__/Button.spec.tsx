import React from 'react';
import { Text } from 'react-native';
import { shallow } from 'enzyme';

import Button from '../Button';

describe('Button shared component', () => {
  // given
  const props = { onPress: jest.fn(), testID: 'button' };
  const child = <Text>Some text</Text>;

  it('should render correctly without children', () => {
    // When
    const wrapper = shallow(<Button {...props} />);

    // Then
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly with a children', () => {
    // When
    const wrapper = shallow(<Button {...props}>{child}</Button>);

    // Then
    expect(wrapper).toMatchSnapshot();
  });

  it('should trigger the onPress props', () => {
    // When
    const wrapper = shallow(<Button {...props}>{child}</Button>);

    wrapper
      .findWhere(node => node.prop('testID') === 'button')
      .simulate('press');

    // Then
    expect(props.onPress.mock.calls.length).toBe(1);
  });
});
