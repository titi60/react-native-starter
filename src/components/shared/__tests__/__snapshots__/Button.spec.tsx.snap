// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button shared component should render correctly with a children 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[MockFunction]}
  testID="button-touchable"
>
  <Text>
    Some text
  </Text>
</TouchableOpacity>
`;

exports[`Button shared component should render correctly without children 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[MockFunction]}
  testID="button-touchable"
/>
`;
